gpMgmt/test/behave/mgmt_utils/gpperfmon.feature

```diff
@@ -90,6 +90,31 @@ Feature: gpperfmon
         Then wait until the results from boolean sql "SELECT count(*) = 0 FROM queries_history WHERE query_text like '--alter distributed by%'" is "true"
         And wait until the results from boolean sql "SELECT count(*) = 1 FROM queries_history WHERE query_text like '--end flag%'" is "true"
 
+    @gpperfmon_query_history
+    Scenario: gpperfmon adds time-consuming queries with different executed statement to queries_now and queries_history
+        Given gpperfmon is configured and running in qamode
+        When the user truncates "queries_history" tables in "gpperfmon"
+        When below sql is executed in "gptest" db
+        """
+            DO $$
+                DECLARE r record;
+            BEGIN
+                FOR r IN
+                    select '
+                        DROP TABLE IF EXISTS test;
+                        CREATE TABLE test(a int, b int) DISTRIBUTED BY (a);
+                        INSERT INTO test VALUES (1, 1);
+                    ' as stmt
+                    where pg_sleep(30) is not null
```

Comment:
30 seconds is not enough, the test passes even without a patch. the minimum time with I was able to reproduce the problem is 70 seconds. if it is possible to reproduce the problem with less time, it would be cool, but for now I suggest this time.

also, do we need all this logic inside the query (drop table, create table and insert)?

Reply:
maybe it's worth trying to run this test in parallel with another one to reduce the total test execution time? perhaps you can change the "get info about current queries" test so that it can work in parallel while a long query from your test is being executed.

Reply:
Files are attempted to be removed when they are older than `time_t cutoff = time(0) - gpmmon_quantum() * 3;` and lose bloom in a rather floating number of `agg_dup()` iterations.
I tried testing it with the following config (which corresponds to the settings in the behave tests except for min_query_time to filter junk):
```
qamode = 1
quantum = 10
harvest_interval = 5
min_query_time = 1
```
And to the surprise it the minimum time to reproduce for me was 80 seconds. Not only that differs from your result, it looks like it doesn't depend on the quantum setting (as with quantum = 10, the 30 secs query should be enough).

The solid number I used when developing was 100 seconds, so yes, short execution times won't work.

The inside query logic is to force execution on segments as the originally discovered problematic query did.

Should try running in parallel.

Reply:
the minimum value for harvest_interval according to the documentation is 30 seconds. I'm not sure if the effect has a smaller value.

Reply:
`qamode = 1` allows to override limits on config parameters, `harvest_interval = 5` is used in behave tests (https://github.com/arenadata/gpdb/blob/d1861f97d9878dc57e4b89f27c17cc2d1aa09f5e/gpMgmt/test/behave/mgmt_utils/steps/mgmt_utils.py#L2534)